handlers_test.go

```diff
@@ -199,6 +199,353 @@ func TestExternalCoordinatorServer(t *testing.T) {
+
+			// Register stale history data.
+			nodeFrom, nodeTo := generateTestKeys(t)
+			failTime := time.Now().Add(-15 * time.Minute).Unix()
+			successTime := time.Now().Add(-12 * time.Minute).Unix()
+			req := &ecrpc.RegisterMissionControlRequest{
+				Pairs: []*ecrpc.PairHistory{
+					{
+						NodeFrom: nodeFrom,
+						NodeTo:   nodeTo,
+						History: &ecrpc.PairData{
+							FailTime:    failTime,
+							SuccessTime: successTime,
+						},
+					},
+				},
+			}
+
+			_, err := server.RegisterMissionControl(
+				context.Background(), req,
+			)
+			require.Error(t, err)
+
+			q_resp, q_err := server.QueryAggregatedMissionControl(
+				context.Background(),
+				&ecrpc.QueryAggregatedMissionControlRequest{},
+			)
+			require.NoError(t, q_err)
+
+			// Check that there are no data in the db cause all
+			// data was stale.
+			require.Empty(t, q_resp.Pairs)
+		})
+
+		// Case 13: Register new pair with old success and fail times
+		// and verify merging works correctly by persisting only the
+		// already existing pair recent times.
+		t.Run("RegisterNewPairWithOldTimeAndMerge", func(t *testing.T) {
+			// Clear the database.
+			err = clearDatabase(db)
+			require.NoError(t, err)
+
+			// Register a pair with current unix time.
+			nodeFrom, nodeTo := generateTestKeys(t)
+			failTime := time.Now().Unix()
+			successTime := time.Now().Unix()
+			req := &ecrpc.RegisterMissionControlRequest{
+				Pairs: []*ecrpc.PairHistory{
+					{
+						NodeFrom: nodeFrom,
+						NodeTo:   nodeTo,
+						History: &ecrpc.PairData{
+							FailTime:       failTime,
+							FailAmtSat:     100,
+							FailAmtMsat:    100_000,
+							SuccessTime:    successTime,
+							SuccessAmtSat:  200,
+							SuccessAmtMsat: 200_000,
+						},
+					},
+				},
+			}
+			resp, err := server.RegisterMissionControl(
+				context.Background(), req,
+			)
+			require.NoError(t, err)
+			require.NotNil(t, resp)
+
+			// Register a new pair with same key but with older
+			// success and failed times.
+			oSuccessTime := time.Now().Add(-5 * time.Minute).Unix()
+			oFailTime := time.Now().Add(-5 * time.Minute).Unix()
+			req = &ecrpc.RegisterMissionControlRequest{
+				Pairs: []*ecrpc.PairHistory{
+					{
+						NodeFrom: nodeFrom,
+						NodeTo:   nodeTo,
+						History: &ecrpc.PairData{
+							FailTime:       oFailTime,
+							FailAmtSat:     100,
+							FailAmtMsat:    100_000,
+							SuccessTime:    oSuccessTime,
+							SuccessAmtSat:  200,
+							SuccessAmtMsat: 200_000,
+						},
+					},
+				},
+			}
+			resp, err = server.RegisterMissionControl(
+				context.Background(), req,
+			)
+			require.NoError(t, err)
+			require.NotNil(t, resp)
+
+			q_resp, q_err := server.QueryAggregatedMissionControl(
+				context.Background(),
+				&ecrpc.QueryAggregatedMissionControlRequest{},
+			)
+			require.NoError(t, q_err)
+			require.NotNil(t, q_resp)
+
+			// Wait a bit to let the database execute the write
+			// batch transactions.
+			time.Sleep(1 * time.Second)
```

Comment:
Maybe we use something similar like: wait.NoError ?

Reply:
I believe we shouldn't rely solely on errors to confirm if the batch write transaction has already executed, given its asynchronous nature. However, one improvement we could implement is setting `config.Database.MaxBatchDelay` to an extremely small value, like `1ns`, eliminating the need for waiting altogether?

Reply:
Updated it to `1ns` and all test cases passed. please let me know if we should go with this approach?

Reply:
That's also a fine approach. I was thinking of something similar like:

```
err = wait.NoError(func() error {
	resp, err := server.QueryAggregatedMissionControl(
				context.Background(),
				&ecrpc.QueryAggregatedMissionControlRequest{},
			)
	         if len(resp.Pairs) == 1 {
			return nil
		}

		return fmt.Errorf("want %v, got %v entries in mission control data: %v",
			1, len(resp.Pairs) )
	}, Timeout)
```

Reply:
Ah this looks cool actually! what is the go package for this `wait.NoError` to be accessible?

Reply:
Its in the lntest/wait package in the lnd repo